mono/mini/mini.c

```diff
@@ -4194,8 +4194,8 @@ mono_jit_compile_method_inner (MonoMethod *method, MonoDomain *target_domain, in
 		char *fullname = mono_method_full_name (method, TRUE);
 		char *msg = g_strdup_printf ("Attempting to JIT compile method '%s' while running with --aot-only. See http://docs.xamarin.com/ios/about/limitations for more information.\n", fullname);
 
-		ex = mono_get_exception_execution_engine (msg);
-		mono_error_set_exception_instance (error, ex);
+		mono_error_set_execution_engine (error, msg);
```

Comment:
`mono_error_set_*` will do a strdup internally, so you can just write `mono_error_set_execution_engine (error, "Attempting to ...", fullname)`